tests/activity_api_reset_test.go

```diff
@@ -0,0 +1,325 @@
+	activityCompleteCh1 <- struct{}{}
+	// wait a bit to make sure previous activity is completed
+	util.InterruptibleSleep(ctx, 1*time.Second)
+
+	// check if workflow and activity are still running
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, enumspb.PENDING_ACTIVITY_STATE_STARTED, description.PendingActivities[0].State)
+		// also verify that the number of attempts was reset
+		assert.Equal(t, int32(1), description.PendingActivities[0].Attempt)
+	}, 5*time.Second, 100*time.Millisecond)
+
+	// let activity finish
+	activityCompleteCh2 <- struct{}{}
+
+	// wait for workflow to complete
+	var out string
+	err = workflowRun.Get(ctx, &out)
+	s.NoError(err)
+}
+
+func (s *ActivityApiResetClientTestSuite) TestActivityResetApi_InRetry() {
+	// reset is called while activity is in retry
+	s.initialRetryInterval = 1 * time.Minute
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	var activityCompleteCount atomic.Int32
+	activityCompleteCh := make(chan struct{})
+
+	activityFunction := func() (string, error) {
+		activityCompleteCount.Add(1)
+
+		if activityCompleteCount.Load() == 1 {
+			activityErr := errors.New("bad-luck-please-retry")
+			return "", activityErr
+		}
+
+		s.WaitForChannel(ctx, activityCompleteCh)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := testcore.RandomizeStr("wf_id-" + s.T().Name())
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        wfId,
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+
+	// wait for activity to start, fail and wait for retry
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, enumspb.PENDING_ACTIVITY_STATE_SCHEDULED, description.PendingActivities[0].State)
+		assert.Equal(t, int32(1), activityCompleteCount.Load())
+	}, 5*time.Second, 200*time.Millisecond)
+
+	resetRequest := &workflowservicepb.ResetActivityByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+		NoWait:     true,
+	}
+	resp, err := s.FrontendClient().ResetActivityById(ctx, resetRequest)
+	s.NoError(err)
+	s.NotNil(resp)
+
+	// wait for activity to start. Wait time is shorter than original retry interval
+	s.EventuallyWithT(func(t *assert.CollectT) {
```

Comment:
Same as above, just check that the attempt was reset to 1 without an eventually block.